test/integration/targets/aws_lambda/tasks/main.yml

```diff
@@ -101,31 +107,26 @@
 
     - name: test state=present - upload the lambda
       lambda:
-        name="{{lambda_function_name}}"
-        runtime="python2.7"
-        handler="mini_lambda.handler"
-        role="ansible_lambda_role"
-        ec2_region='{{ec2_region}}'
-        ec2_access_key='{{ec2_access_key}}'
-        ec2_secret_key='{{ec2_secret_key}}'
-        security_token='{{security_token}}'
-        zip_file="{{zip_res.dest}}"
+        name: "{{lambda_function_name}}"
+        runtime: "python2.7"
+        handler: "mini_lambda.handler"
+        role: "ansible_lambda_role"
+        zip_file: "{{zip_res.dest}}"
+        <<: *aws_connection_info
       register: result
 
     - name: assert lambda upload succeeded
       assert:
         that:
-           - 'result is not failed'
+          - result is not failed
+          - result.configuration.tracing_config.mode == "PassThrough"
 
     - name: test lambda works
       execute_lambda:
         name: "{{lambda_function_name}}"
         payload:
           name: "Mr Ansible Tests"
-        ec2_region: '{{ec2_region}}'
-        ec2_access_key: '{{ec2_access_key}}'
-        ec2_secret_key: '{{ec2_secret_key}}'
-        security_token: '{{security_token}}'
+        <<: *aws_connection_info
```

Comment:
The YAML anchors should also be removed from the file now that it is no longer defined. `module_defaults` sets the default parameters for the modules in the `aws` group so these will all get the credentials and you should only set them on the task if you want to override the defaults. There are a number of tasks that expect `region` to not be set. Those tasks need `region:` added to them like I had in my commit. Sorry, this is why I was just going to push to the branch. Fiddly details.

Reply:
yeah, my bad. really just should have cherry-picked your commit, though it would be easier this way :P